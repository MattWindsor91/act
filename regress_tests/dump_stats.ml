(* The Automagic Compiler Tormentor

   Copyright (c) 2018--2020 Matt Windsor and contributors

   ACT itself is licensed under the MIT License. See the LICENSE file in the
   project root for more information.

   ACT is based in part on code from the Herdtools7 project
   (https://github.com/herd/herdtools7) : see the LICENSE.herd file in the
   project root for more information. *)

(** Tests the output of the 'act-c dump-stats' command. *)

open Core

let dump_stats_of_file ~(file : Fpath.t) ~(path : Fpath.t) : unit Or_error.t
    =
  ignore file ;
  Act_litmus_c.Dump_stats.Filter.run ()
    (Plumbing.Input.of_fpath path)
    Plumbing.Output.stdout

let run (test_dir : Fpath.t) : unit Or_error.t =
  let full_dir = Fpath.(test_dir / "litmus" / "") in
  Common.regress_on_files "Dump-stats" ~dir:full_dir ~ext:"litmus"
    ~f:dump_stats_of_file

let command =
  Common.make_regress_command ~summary:"runs dump-stats regressions" run

let () = Command.run command
