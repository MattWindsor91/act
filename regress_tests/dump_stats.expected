# Dump-stats tests

## fuzz_local_bools.c.litmus

```
threads 3
returns 1
literals.bool 1
atomics.expression.cmpxchg 0
atomics.expression.fence 0
atomics.expression.fetch 5
atomics.expression.load 3
atomics.expression.store 0
atomics.expression.xchg 0
mem-orders.expression.memory_order_relaxed 2
mem-orders.expression.memory_order_consume 1
mem-orders.expression.memory_order_acquire 1
mem-orders.expression.memory_order_release 0
mem-orders.expression.memory_order_acq_rel 1
mem-orders.expression.memory_order_seq_cst 3
atomics.statement.cmpxchg 0
atomics.statement.fence 1
atomics.statement.fetch 5
atomics.statement.load 0
atomics.statement.store 4
atomics.statement.xchg 0
mem-orders.statement.memory_order_relaxed 5
mem-orders.statement.memory_order_consume 2
mem-orders.statement.memory_order_acquire 0
mem-orders.statement.memory_order_release 1
mem-orders.statement.memory_order_acq_rel 0
mem-orders.statement.memory_order_seq_cst 7
```
## iriw.c.litmus

```
threads 4
returns 0
literals.bool 0
atomics.expression.cmpxchg 0
atomics.expression.fence 0
atomics.expression.fetch 0
atomics.expression.load 4
atomics.expression.store 0
atomics.expression.xchg 0
mem-orders.expression.memory_order_relaxed 0
mem-orders.expression.memory_order_consume 0
mem-orders.expression.memory_order_acquire 2
mem-orders.expression.memory_order_release 0
mem-orders.expression.memory_order_acq_rel 0
mem-orders.expression.memory_order_seq_cst 2
atomics.statement.cmpxchg 0
atomics.statement.fence 0
atomics.statement.fetch 0
atomics.statement.load 0
atomics.statement.store 2
atomics.statement.xchg 0
mem-orders.statement.memory_order_relaxed 0
mem-orders.statement.memory_order_consume 0
mem-orders.statement.memory_order_acquire 0
mem-orders.statement.memory_order_release 0
mem-orders.statement.memory_order_acq_rel 0
mem-orders.statement.memory_order_seq_cst 2
```
## iriw_fuzz_redundant.c.litmus

```
threads 5
returns 0
literals.bool 0
atomics.expression.cmpxchg 0
atomics.expression.fence 0
atomics.expression.fetch 0
atomics.expression.load 4
atomics.expression.store 0
atomics.expression.xchg 0
mem-orders.expression.memory_order_relaxed 0
mem-orders.expression.memory_order_consume 0
mem-orders.expression.memory_order_acquire 2
mem-orders.expression.memory_order_release 0
mem-orders.expression.memory_order_acq_rel 0
mem-orders.expression.memory_order_seq_cst 2
atomics.statement.cmpxchg 0
atomics.statement.fence 0
atomics.statement.fetch 0
atomics.statement.load 0
atomics.statement.store 5
atomics.statement.xchg 0
mem-orders.statement.memory_order_relaxed 1
mem-orders.statement.memory_order_consume 0
mem-orders.statement.memory_order_acquire 0
mem-orders.statement.memory_order_release 0
mem-orders.statement.memory_order_acq_rel 0
mem-orders.statement.memory_order_seq_cst 4
```
## memalloy_cmpxchg.c.litmus

```
threads 2
returns 0
literals.bool 0
atomics.expression.cmpxchg 0
atomics.expression.fence 0
atomics.expression.fetch 0
atomics.expression.load 0
atomics.expression.store 0
atomics.expression.xchg 0
mem-orders.expression.memory_order_relaxed 0
mem-orders.expression.memory_order_consume 0
mem-orders.expression.memory_order_acquire 0
mem-orders.expression.memory_order_release 0
mem-orders.expression.memory_order_acq_rel 0
mem-orders.expression.memory_order_seq_cst 0
atomics.statement.cmpxchg 1
atomics.statement.fence 0
atomics.statement.fetch 0
atomics.statement.load 0
atomics.statement.store 1
atomics.statement.xchg 0
mem-orders.statement.memory_order_relaxed 3
mem-orders.statement.memory_order_consume 0
mem-orders.statement.memory_order_acquire 0
mem-orders.statement.memory_order_release 0
mem-orders.statement.memory_order_acq_rel 0
mem-orders.statement.memory_order_seq_cst 0
```
## memalloy_na_int.c.litmus

```
threads 1
returns 0
literals.bool 0
atomics.expression.cmpxchg 0
atomics.expression.fence 0
atomics.expression.fetch 0
atomics.expression.load 0
atomics.expression.store 0
atomics.expression.xchg 0
mem-orders.expression.memory_order_relaxed 0
mem-orders.expression.memory_order_consume 0
mem-orders.expression.memory_order_acquire 0
mem-orders.expression.memory_order_release 0
mem-orders.expression.memory_order_acq_rel 0
mem-orders.expression.memory_order_seq_cst 0
atomics.statement.cmpxchg 0
atomics.statement.fence 0
atomics.statement.fetch 0
atomics.statement.load 0
atomics.statement.store 0
atomics.statement.xchg 0
mem-orders.statement.memory_order_relaxed 0
mem-orders.statement.memory_order_consume 0
mem-orders.statement.memory_order_acquire 0
mem-orders.statement.memory_order_release 0
mem-orders.statement.memory_order_acq_rel 0
mem-orders.statement.memory_order_seq_cst 0
```
## rwc.c.litmus

```
threads 3
returns 0
literals.bool 0
atomics.expression.cmpxchg 0
atomics.expression.fence 0
atomics.expression.fetch 0
atomics.expression.load 3
atomics.expression.store 0
atomics.expression.xchg 0
mem-orders.expression.memory_order_relaxed 0
mem-orders.expression.memory_order_consume 0
mem-orders.expression.memory_order_acquire 1
mem-orders.expression.memory_order_release 0
mem-orders.expression.memory_order_acq_rel 0
mem-orders.expression.memory_order_seq_cst 2
atomics.statement.cmpxchg 0
atomics.statement.fence 0
atomics.statement.fetch 0
atomics.statement.load 0
atomics.statement.store 2
atomics.statement.xchg 0
mem-orders.statement.memory_order_relaxed 0
mem-orders.statement.memory_order_consume 0
mem-orders.statement.memory_order_acquire 0
mem-orders.statement.memory_order_release 0
mem-orders.statement.memory_order_acq_rel 0
mem-orders.statement.memory_order_seq_cst 2
```

Ran 6 test(s).
